### ISSUE TITLE
Hide Server: openresty from HTTP headers

### ISSUE THREAD

**Issue by oxismailxo** [NginxProxyManager/nginx-proxy-manager#1061]
Hello NPM community,

is it possible to hide Server: openresty from the HTTP headers? if yes how?, please

![image](https://user-images.githubusercontent.com/68305651/116990571-d4dfc600-acc2-11eb-9446-e87eff203780.png)

Thank you in advance .

**Comment by l4rm4nd** [comment]
Usually one would modify the ``/etc/nginx/nginx.conf`` and do

````
server_tokens                 "";
````

However, this only works for commercial nginx versions and not for nginx proxy manager. You would have to compile from source and patch the disclosed version information. Alternatively, you would have to install additional modules like ``nginx-extras``, which allows stuff like this.

Note that hiding server banners is security through obscurity. You should rather focus on a proper patch management process to keep your assets updated with the most recent patch releases. Disclosing ``openresty`` in a server's HTTP response does not introduce any security issues.

Currently, the configuration specifies a best practice directive such

````
server_tokens                 off;
````

This successfully hides the exact version number of your web service. This is sufficient from a security perspective.

**Comment by oxismailxo** [comment]
Alright @l4rm4nd, thank you so much for the clarification & the detailed answer!

I appreciate your help.

**Comment by jc21** [comment]
Great response @l4rm4nd

FYI openresty is [compiled for NPM here](https://github.com/jc21/docker-nginx-full/blob/master/scripts/build-openresty) should anyone feel like suggesting tweaks in future.

**Comment by l4rm4nd** [comment]
> Great response @l4rm4nd
> 
> FYI openresty is [compiled for NPM here](https://github.com/jc21/docker-nginx-full/blob/master/scripts/build-openresty) should anyone feel like suggesting tweaks in future.

So if one really wants to alter the disclosed ``openresty`` product name, it should be possible via an additional build option in [build-openresty](https://github.com/jc21/docker-nginx-full/blob/master/scripts/build-openresty). Just define your custom name like ``MyCustomServerName`` and specify it for the ``./configure`` call:

````
./configure \
       ...
        --build=MyCustomServerName \
       ...
````

**Comment by jc21** [comment]
Yeah it's possible. You can open a PR on the docker-nginx-full project to test that and CI will build it for you. Beware that builds take over an hour with all the architectures to build for.
